### Загрузка german_real_estate_data

In [2]:
import pandas as pd

In [6]:
ger_est = pd.read_csv('german_real_estate_data.csv')
ger_est.info()
ger_est.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25613 entries, 0 to 25612
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   address         25613 non-null  object
 1   property_id     25613 non-null  object
 2   square_meters   25613 non-null  int64 
 3   total_floors    25613 non-null  int64 
 4   floor           25613 non-null  int64 
 5   type            25613 non-null  object
 6   price           25613 non-null  int64 
 7   days_on_market  25613 non-null  int64 
 8   rooms           25613 non-null  int64 
 9   city            25613 non-null  object
dtypes: int64(6), object(4)
memory usage: 2.0+ MB


address property_id  square_meters  total_floors  floor  \
0   250 High St, Berlin    ID012715             20            29     17   
1   225 High St, Berlin    ID017686             20            17      1   
2    29 High St, Berlin    ID015698             20            22      5   
3  179 Broadway, Berlin    ID017078             20            21     25   
4   60 Broadway, Berlin    ID020426             20             9     27   

          type    price  days_on_market  rooms    city  
0  Residential  2966921             115      1  Berlin  
1  Residential  3259048             247      5  Berlin  
2   Commercial  7071926             167      5  Berlin  
3   Commercial  7759168             266      9  Berlin  
4  Residential  8320308             288      5  Berlin

In [9]:
ger_est['type'].unique()

array(['Residential', 'Commercial'], dtype=object)

Файл содержит данные о недвижимости в Германии с 25,613 записями и 10 столбцами. Вот основные столбцы:

address: Адрес недвижимости.

property_id: Идентификатор недвижимости.

square_meters: Площадь в квадратных метрах.

total_floors: Общее количество этажей в здании.

floor: Этаж, на котором находится недвижимость.

type: Тип недвижимости (Residential или Commercial).

price: Цена недвижимости.

days_on_market: Количество дней на рынке.

rooms: Количество комнат.

city: Город.

### Исследование недвижимости Германии

Ниже сформулированы целевые исследования по рынку недвижимости Германии 

1 Анализ цен на недвижимость: Средняя цена за квадратный метр по городам. Сравнение цен на коммерческую и жилую недвижимость. Влияние количества этажей или комнат на цену.

2 Анализ времени нахождения на рынке: Исследование средней продолжительности продажи недвижимости в зависимости от типа, города и других характеристик. Влияние цены на количество дней на рынке.

3 Географический анализ: Сравнение стоимости недвижимости между различными городами. Поиск зависимостей между локацией (адресом) и стоимостью.



#### 1 Анализ цен на недвижимость

In [72]:
ger_est['price_per_sqm'] = ger_est['price'] / ger_est['square_meters']
avg_city = ger_est.groupby('city')['price_per_sqm'].mean().sort_values(ascending=False)
avg_city

city
Dortmund      35897.679200
Hamburg       35381.540470
Frankfurt     35173.673783
Cologne       34534.409536
Stuttgart     34161.264777
Berlin        34150.988514
Leipzig       34085.164736
Essen         33840.400245
Munich        33658.770241
Düsseldorf    33433.036184
Name: price_per_sqm, dtype: float64

In [54]:
avg_type = ger_est.groupby('type')['price_per_sqm'].mean()
avg_type

type
Commercial     34784.820162
Residential    34082.249541
Name: price_per_sqm, dtype: float64

In [56]:
avg_floors = ger_est.groupby('total_floors')['price_per_sqm'].mean()
avg_rooms = ger_est.groupby('rooms')['price_per_sqm'].mean()

avg_floors, avg_rooms

(total_floors
 1     35021.080618
 2     35691.942538
 3     32894.369680
 4     32247.495627
 5     35383.852570
 6     32591.993096
 7     35775.214070
 8     35630.746895
 9     33840.595978
 10    34435.250302
 11    35677.821621
 12    32848.929492
 13    33990.950996
 14    37199.027762
 15    35623.573292
 16    34374.324176
 17    34459.831669
 18    33329.578571
 19    33199.497621
 20    30099.649125
 21    35045.878506
 22    36420.832412
 23    33895.365364
 24    36572.198901
 25    34444.235329
 26    35506.448014
 27    33376.394059
 28    35218.898934
 29    34254.616861
 Name: price_per_sqm, dtype: float64,
 rooms
 1    34591.946372
 2    34177.756296
 3    32641.547755
 4    34577.751828
 5    33447.268983
 6    36536.423581
 7    33767.643297
 8    35244.113701
 9    34881.335259
 Name: price_per_sqm, dtype: float64)

Средняя цена за квадратный метр по городам:
Наиболее дорогие города (средняя цена за квадратный метр):
Dortmund: ~35,897 €/м²
Hamburg: ~35,381 €/м²
Frankfurt: ~35,174 €/м²
Наименее дорогие города:
Düsseldorf: ~33,433 €/м²
Munich: ~33,659 €/м²

Сравнение цен на коммерческую и жилую недвижимость:
Коммерческая недвижимость: ~34,785 €/м²
Жилая недвижимость: ~34,082 €/м²
Коммерческая недвижимость в среднем немного дороже жилой.

Влияние количества этажей на цену:
Средняя цена за квадратный метр варьируется в зависимости от общего количества этажей в здании. Например:
1 этаж: ~35,021 €/м²
14 этажей: ~37,199 €/м² (максимальное значение)
20 этажей: ~30,099 €/м² (минимальное значение)

Влияние количества комнат на цену:
Средняя цена за квадратный метр в зависимости от количества комнат:
1 комната: ~34,592 €/м²
6 комнат: ~36,536 €/м² (максимальное значение)
3 комнаты: ~32,641 €/м² (минимальное значение)

#### 2 Анализ времени нахождения на рынке

In [57]:
avg_days_type = ger_est.groupby('type')['days_on_market'].mean()
avg_days_city = ger_est.groupby('city')['days_on_market'].mean().sort_values(ascending=False)
print('Среднее количество дней на рынке для недвижимости в зависимости от',avg_days_type,
'\n','Среднее количество дней на рынке для недвижимости в зависимости от', avg_days_city)

Среднее количество дней на рынке для недвижимости в зависимости от type
Commercial     182.332343
Residential    181.687797
Name: days_on_market, dtype: float64 
 Среднее количество дней на рынке для недвижимости в зависимости от city
Frankfurt     183.704981
Berlin        183.596957
Cologne       182.770760
Düsseldorf    182.554335
Hamburg       182.241248
Essen         181.811873
Dortmund      181.655547
Munich        181.410246
Stuttgart     180.235986
Leipzig       180.088540
Name: days_on_market, dtype: float64


In [58]:
corr_days = ger_est['price'].corr(ger_est['days_on_market'])
corr_days

0.009578899552588303

Среднее количество дней на рынке для недвижимости разного типа:
Коммерческая недвижимость: в среднем 182 дня.
Жилая недвижимость: в среднем 181 день.
Время нахождения на рынке для обоих типов недвижимости примерно одинаково.

Среднее количество дней на рынке по городам:
Город с самым долгим средним временем на рынке:
Frankfurt: ~184 дня.
Город с самым коротким временем на рынке:
Leipzig: ~180 дней.
Влияние цены на количество дней на рынке:

Корреляция между ценой и количеством дней на рынке составляет 0.0096, что указывает на очень слабую положительную корреляцию. Это означает, что цена практически не влияет на количество дней на рынке.

#### 3 Географический анализ

In [66]:
average_city = ger_est.groupby('city')['price'].mean().sort_values(ascending=False)
average_city

city
Leipzig       5.108872e+06
Hamburg       5.077305e+06
Dortmund      5.075284e+06
Stuttgart     5.071732e+06
Düsseldorf    5.070984e+06
Cologne       5.051072e+06
Berlin        5.045404e+06
Frankfurt     5.014324e+06
Essen         4.996852e+06
Munich        4.984504e+06
Name: price, dtype: float64

In [68]:
ger_est['street'] = ger_est['address'].apply(lambda x: x.split(',')[0].strip())
average_street = ger_est.groupby('street')['price'].mean().sort_values(ascending=False)
average_street

street
161 Park Ave    6.790178e+06
83 Broadway     6.761824e+06
40 Broadway     6.701696e+06
160 Broadway    6.626970e+06
227 High St     6.613109e+06
                    ...     
164 High St     3.408272e+06
20 Park Ave     3.386043e+06
79 Park Ave     3.354149e+06
59 Main St      3.130725e+06
184 Broadway    2.984306e+06
Name: price, Length: 1200, dtype: float64

 Средняя стоимость недвижимости по городам:
(Leipzig) имеет самую высокую среднюю стоимость недвижимости (около 5.11 млн евро).
(Hamburg) и (Dortmund) также находятся среди городов с высокой средней стоимостью (около 5.07 млн евро).
(Munich), известный как один из самых дорогих городов Германии, оказывается в конце списка, что может указывать на разные типы недвижимости или специфику выборки.

Средняя стоимость недвижимости по улицам:
Улица 161 Park Ave имеет самую высокую среднюю стоимость недвижимости (около 6.79 млн евро).
Несколько улиц на Бродвее (например, 83 Broadway, 40 Broadway) также отличаются высокой средней стоимостью недвижимости (более 6.7 млн евро).
Разница в средней цене между самыми дорогими и менее дорогими улицами значительна. Например, 184 Broadway имеет среднюю стоимость в районе 2.98 млн евро, что гораздо ниже по сравнению с лидерами.

Общие выводы:
Разброс цен между городами относительно небольшой, все города находятся примерно в одном диапазоне (около 5 млн евро). Это может указывать на схожую ценовую политику или специфику предложений в этих городах.
Разброс цен по улицам гораздо более значителен. Улицы, такие как Park Ave и Broadway, выделяются более высокой стоимостью, что, вероятно, связано с их престижностью или расположением в центральных частях городов.

### Вывод 

Эти результаты могут быть полезны для дальнейшего анализа рынка недвижимости, например, при оценке инвестиционной привлекательности разных районов и улиц.